cmd/deploy_test.go

```diff
@@ -40,6 +41,19 @@ const ExampleAPIModelWithDNSPrefix = `{
   }
   `
 
+const ExampleAPIModelWithoutServicePrincipalProfile = `{
```

Comment:
Let's also add a test case for empty `ServicePrincipalProfile`

Reply:
1. That's correct
2. Yes, exactly. It shouldn't be very complicated to add to this PR. I believe all you need to do is to change your if statement in deploy.go to `} else if (dc.containerService.Properties.ServicePrincipalProfile == nil || (dc.containerService.Properties.ServicePrincipalProfile.ClientID == ""  && dc.containerService.Properties.ServicePrincipalProfile.ClientSecret == "") ) && dc.ClientID.String() != "" && dc.ClientSecret != "" ` and add the appropriate unit test case. We just want to make sure that the change you made applies if the sp profile is declared but contains no ID and secret values.

Reply:
Thanks. Makes sense. So test cases for conditions where id and secret are specified in both the api model and deployment command can be taken up later.

Reply:
@CecileRobertMichon i have made the requested changes, please review and let me know if any further changes are needed